File: lib/core/models/doctor.dart
// Doctor model — represents a doctor available for consultation
class Doctor {
  final String id;
  final String name;
  final String speciality;
  final String qualification;
  final String facility;
  final String experience; // e.g. "8 years"
  final String availableSlots;
  final bool acceptsOnline;

  const Doctor({
    required this.id,
    required this.name,
    required this.speciality,
    required this.qualification,
    required this.facility,
    required this.experience,
    required this.availableSlots,
    required this.acceptsOnline,
  });

  factory Doctor.fromJson(Map<String, dynamic> json) {
    return Doctor(
      id: json['id'] as String? ?? '',
      name: json['name'] as String? ?? '',
      speciality: json['speciality'] as String? ?? '',
      qualification: json['qualification'] as String? ?? '',
      facility: json['facility'] as String? ?? '',
      experience: json['experience'] as String? ?? '',
      availableSlots: json['availableSlots'] as String? ?? '',
      acceptsOnline: json['acceptsOnline'] as bool? ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'speciality': speciality,
      'qualification': qualification,
      'facility': facility,
      'experience': experience,
      'availableSlots': availableSlots,
      'acceptsOnline': acceptsOnline,
    };
  }
}

File: backend/src/models/Doctor.js
// Mongoose model for Doctors
const mongoose = require('mongoose');

const doctorSchema = new mongoose.Schema(
    {
        name: {
            type: String,
            required: true,
            trim: true,
            index: true,
        },
        speciality: {
            type: String,
            required: true,
            trim: true,
            index: true,
        },
        qualification: {
            type: String,
            default: 'MBBS',
        },
        facilityId: {
            type: mongoose.Schema.Types.ObjectId,
            ref: 'Facility',
            default: null,
            index: true,
        },
        facilityName: {
            type: String,
            default: '',
        },
        experience: {
            type: String,
            default: '5 years',
        },
        availableSlots: {
            type: String,
            default: 'Mon–Fri: 10am–2pm',
        },
        acceptsOnline: {
            type: Boolean,
            default: false,
            index: true,
        },
        phone: {
            type: String,
            default: '',
        },
        isActive: {
            type: Boolean,
            default: true,
        },
    },
    {
        timestamps: true,
        collection: 'doctors',
    }
);

doctorSchema.index({ name: 'text', speciality: 'text', facilityName: 'text' });

module.exports = mongoose.model('Doctor', doctorSchema);
